### PR TITLE
Refactor: Extract test configuration

### PR DESCRIPTION
**Pull request by lwojtow**

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -121,7 +121,15 @@ The test index makes use of symbolic language in describing connection and messa
 
 ### ZG-CONFORMANCE-006
 
-    The node requests mtGET_PEER_SHARD_INFO_V2 after start up.
+    The node requests mtGET_PEER_SHARD_INFO_V2 after connection and handshake.
 
+    <>
+    <-> handshake
```
Comment (ljedrz) [inline SPEC.md:127]:
```suggestion
```
`<>` implies a complete handshake

Comment (lwojtow) [reply SPEC.md:127]: I thought this only implies connection. How to represent 'no handshake'?

Comment (lwojtow) [reply SPEC.md]: Done

File: src/tools/config.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+use std::io;
+
+use crate::{protocol::codecs::binary::Payload, setup::config::NodeConfig};
+
+/// Test configuration. Contains setup options for [SyntheticNode], [Node] and [pea2pea::Config].
```
Comment (ljedrz) [inline src/tools/config.rs:5]: have you checked that `SyntheticNode` and `Node` are linked correctly when using `cargo doc`? I don't see them in the scope

File: src/tests/conformance/query/peer_shard_info.rs [hunk 3]
```diff
@@ -3,12 +3,17 @@
...
 
 #[tokio::test]
-async fn node_should_query_for_shard_info() {
+async fn node_should_query_for_shard_info_after_handshake() {
     // ZG-CONFORMANCE-006
     let response_check =
         |m: &BinaryMessage| matches!(&m.payload, Payload::TmGetPeerShardInfoV2(..));
-    perform_response_test(None, &response_check).await;
+    perform_response_test(
+        TestConfig::new().unwrap().with_handshake(true),
```
Comment (ljedrz) [inline src/tests/conformance/query/peer_shard_info.rs:15]: this can just be `Default::default()`

File: src/tools/synth_node.rs [hunk 4]
```diff
@@ -26,22 +27,20 @@ pub struct SyntheticNode {
...
-    /// Starts node bound to localhost and with default [pea2pea::Config]
-    pub async fn start() -> Result<Self, AddrParseError> {
-        let node_config = pea2pea::Config {
-            listener_ip: Some("127.0.0.1".parse()?),
-            ..Default::default()
-        };
-        Ok(Self::new(node_config).await)
+    /// Starts node
+    pub async fn start(config: &TestConfig) -> Self {
+        Self::new(config).await
```
Comment (ljedrz) [inline src/tools/synth_node.rs:43]: doesn't seem like this function is needed, it does the same thing as `new`; one of them needs to go

File: src/tools/synth_node.rs [hunk 5]
```diff
@@ -26,22 +27,20 @@ pub struct SyntheticNode {
...
+        }
         inner.enable_reading().await;
         inner.enable_writing().await;
         Self { inner, receiver }
     }
 
-    /// Starts node bound to localhost and with default [pea2pea::Config]
-    pub async fn start() -> Result<Self, AddrParseError> {
-        let node_config = pea2pea::Config {
-            listener_ip: Some("127.0.0.1".parse()?),
```
Comment (ljedrz) [inline src/tools/synth_node.rs:41]: is this property preserved?

File: src/tools/config.rs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    pub fn with_initial_message(mut self, payload: Payload) -> Self {
+        self.synth_node_config.initial_message = Some(payload);
+        self
+    }
+}
+
+pub struct SyntheticNodeTestConfig {
+    /// Whether or not to call `enable_handshake` when creating a new
```
Comment (ljedrz) [inline src/tools/config.rs:33]: a new node, I presume?

File: SPEC.md [hunk 7]
```diff
@@ -121,7 +121,12 @@ The test index makes use of symbolic language in describing connection and messa
...
 
-    ->
-    <- mtGET_PEER_SHARD_INFO_V2
\ No newline at end of file
+    <>
+    <- mtGET_PEER_SHARD_INFO_V2
+
+### ZG-CONFORMANCE-007
+
+    The node should *NOT* request mtGET_PEER_SHARD_INFO_V2 after connection if no handshake.
```
Comment (ljedrz) [inline SPEC.md:131]:
```suggestion
    The node should *NOT* request mtGET_PEER_SHARD_INFO_V2 after connection if there was no handshake.
```

File: SPEC.md [hunk 8]
```diff
@@ -121,7 +121,12 @@ The test index makes use of symbolic language in describing connection and messa
...
+    The test waits for predefined timeout ensuring no such messages were received.
```
Comment (ljedrz) [inline SPEC.md:132]:
```suggestion
    The test waits for the predefined amount of time, ensuring no such messages were received.
```

File: src/tests/conformance/query/peer_shard_info.rs [hunk 9]
```diff
@@ -3,12 +3,29 @@
...
```
Comment (ljedrz) [inline src/tests/conformance/query/peer_shard_info.rs:15]:
```suggestion
        Default::default(),
```

File: src/tools/synth_node.rs (hunk 5 above)
Comment (ljedrz) [reply src/tools/synth_node.rs:41]: it is in `pea2pea`, but have you checked in this repo's tests? I have doubts :detective:

Comment (lwojtow) [reply src/tools/synth_node.rs:41]: Is this what you mean:
```
impl TestConfig {
    pub fn new() -> io::Result<Self> {
        Ok(Self {
            real_node_config: NodeConfig::new()?,
            synth_node_config: Default::default(),
            pea2pea_config: pea2pea::Config {
                listener_ip: Some("127.0.0.1".parse().unwrap()),
                ..Default::default()
            },
        })
    }

```

File: src/tests/conformance/query/peer_shard_info.rs (hunk 9 above)
Comment (lwojtow) [reply src/tests/conformance/query/peer_shard_info.rs]: Done.

**Review by ljedrz — APPROVED** [review]
LGTM, @niklaslong?

**Pull request merged**